src/helpers/myComponent.js

```diff
@@ -0,0 +1,34 @@
+// import { useMapEvent } from 'react-leaflet/hooks';
+
+// export function MyComponent({ geoInfo }) {
+//   const map = useMapEvent('moveend', () => {
+//     if (geoInfo) {
+//       map.setView(geoInfo, map.getZoom());
+//     }
+//   });
+
+//   return null;
+// }
+
+// export function MyComponent({ geoInfo }) {
+//   const map = useMapEvent(() => {
+//     map.setView(geoInfo.centre.getZoom());
+//   });
+
+//   return null;
+// }
+
+import { useEffect } from 'react';
+import { useMap } from 'react-leaflet';
+
+export function MyComponent({ geoInfo }) {
+  const map = useMap();
+
+  useEffect(() => {
+    if (geoInfo) {
+      map.setView(geoInfo, map.getZoom());
+    }
+  }, [geoInfo, map]);
```

Comment:
What doesn't work for you here? As I can see you haven't said anything about animation here.
If you look at `mat.setView` function declaration you see the third parameter which is an object with 4 possible properties:
```
    animate?: boolean | undefined;
    duration?: number | undefined;
    easeLinearity?: number | undefined;
    noMoveStart?: boolean | undefined;
```

Reply:
I used map.flyTo